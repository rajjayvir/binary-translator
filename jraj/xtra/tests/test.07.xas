  std
  jmp start
foo:
  loadi 2, r0
  ret
start:
  loadi 1, r0
  call foo
  loadi 3, r0

.literal 0


