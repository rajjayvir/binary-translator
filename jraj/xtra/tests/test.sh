#!/bin/sh

if [ ! -x xas ]; then
  make xas 
fi

echo ======================================================
echo ====================== TEST $1 ========================
echo ======================================================
./xas tests/test.$1.xas tests/test.$1.xo > /dev/null
if timeout 10 ./$2/$3 tests/test.$1.xo > tests/test.$1.s; then 
   echo Translated tests/test.$1.s 
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: the program crashed
  echo Exit code $?
  exit 1
fi

echo Compiling runner
gcc -o runner runner.c regsdump.s tests/test.$1.s
if [ ! -x runner ]; then
  echo runner not created.
  exit 1
fi

echo Running runner
if timeout 10 ./runner > tests/test.$1.out; then 
  if diff -w tests/test.$1.out tests/test.$1.expected > /dev/null; then
    echo PASSED
  else
    echo FAILED
    echo ======
    echo ____Your_output____ __Expected_Output___
   #echo +++++++++++++++++++ ++++++++++++++++++++
    diff -y -W 80 tests/test.$1.out tests/test.$1.expected 
    echo =====================
    echo Your Output:
    echo ++++++++++++++++
    cat tests/test.$1.out
    echo =====================
    echo Expected Output:
    echo ++++++++++++++++
    cat tests/test.$1.expected
    exit 1
  fi
elif [ $? -eq 124 ]; then
  echo TIMEOUT
  exit 1
else 
  echo Abnormal program termination: runner crashed
  echo Exit code $?
  exit 1
fi
