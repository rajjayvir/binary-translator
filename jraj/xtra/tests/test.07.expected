Registers:
  %rax: 0x0000000000000000    %rbx: 0x0000000000000000    
  %rcx: 0x0000000000000000    %rdx: 0x0000000000000000    
  %rsi: 0x0000000000000000    %rdi: 0x0000000000000000    
  %r8 : 0x0000000000000000    %r9 : 0x0000000000000000    
  %r10: 0x0000000000000000    %r11: 0x0000000000000000    
  %r12: 0x0000000000000000    %r13: 0x0000000000000000    
  %r14: 0x0000000000000000    %r15: 0x0000000000000000    
Registers:
  %rax: 0x0000000000000000    %rbx: 0x0000000000000000    
  %rcx: 0x0000000000000000    %rdx: 0x0000000000000000    
  %rsi: 0x0000000000000000    %rdi: 0x0000000000000000    
  %r8 : 0x0000000000000000    %r9 : 0x0000000000000000    
  %r10: 0x0000000000000000    %r11: 0x0000000000000000    
  %r12: 0x0000000000000000    %r13: 0x0000000000000000    
  %r14: 0x0000000000000000    %r15: 0x0000000000000000    
Registers:
  %rax: 0x0000000000000001    %rbx: 0x0000000000000000    
  %rcx: 0x0000000000000000    %rdx: 0x0000000000000000    
  %rsi: 0x0000000000000000    %rdi: 0x0000000000000000    
  %r8 : 0x0000000000000000    %r9 : 0x0000000000000000    
  %r10: 0x0000000000000000    %r11: 0x0000000000000000    
  %r12: 0x0000000000000000    %r13: 0x0000000000000000    
  %r14: 0x0000000000000000    %r15: 0x0000000000000000    
Registers:
  %rax: 0x0000000000000001    %rbx: 0x0000000000000000    
  %rcx: 0x0000000000000000    %rdx: 0x0000000000000000    
  %rsi: 0x0000000000000000    %rdi: 0x0000000000000000    
  %r8 : 0x0000000000000000    %r9 : 0x0000000000000000    
  %r10: 0x0000000000000000    %r11: 0x0000000000000000    
  %r12: 0x0000000000000000    %r13: 0x0000000000000000    
  %r14: 0x0000000000000000    %r15: 0x0000000000000000    
Registers:
  %rax: 0x0000000000000002    %rbx: 0x0000000000000000    
  %rcx: 0x0000000000000000    %rdx: 0x0000000000000000    
  %rsi: 0x0000000000000000    %rdi: 0x0000000000000000    
  %r8 : 0x0000000000000000    %r9 : 0x0000000000000000    
  %r10: 0x0000000000000000    %r11: 0x0000000000000000    
  %r12: 0x0000000000000000    %r13: 0x0000000000000000    
  %r14: 0x0000000000000000    %r15: 0x0000000000000000    
Registers:
  %rax: 0x0000000000000002    %rbx: 0x0000000000000000    
  %rcx: 0x0000000000000000    %rdx: 0x0000000000000000    
  %rsi: 0x0000000000000000    %rdi: 0x0000000000000000    
  %r8 : 0x0000000000000000    %r9 : 0x0000000000000000    
  %r10: 0x0000000000000000    %r11: 0x0000000000000000    
  %r12: 0x0000000000000000    %r13: 0x0000000000000000    
  %r14: 0x0000000000000000    %r15: 0x0000000000000000    
Registers:
  %rax: 0x0000000000000003    %rbx: 0x0000000000000000    
  %rcx: 0x0000000000000000    %rdx: 0x0000000000000000    
  %rsi: 0x0000000000000000    %rdi: 0x0000000000000000    
  %r8 : 0x0000000000000000    %r9 : 0x0000000000000000    
  %r10: 0x0000000000000000    %r11: 0x0000000000000000    
  %r12: 0x0000000000000000    %r13: 0x0000000000000000    
  %r14: 0x0000000000000000    %r15: 0x0000000000000000    
